src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/HttpApi/ApiClientExtensions.cs

```diff
@@ -31,6 +34,45 @@ public static async Task<IEnumerable<ProcessIdentifier>> GetProcessesAsync(this
             return await client.GetProcessesAsync(timeoutSource.Token).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// GET /processes with retry attempts
+        /// </summary>
+        public static async Task<IEnumerable<ProcessIdentifier>> GetProcessesWithRetryAsync(this ApiClient client, ITestOutputHelper outputHelper, Func<ProcessIdentifier, bool> filter = null, int maxAttempts = 5)
+        {
+            IList<ProcessIdentifier> identifiers = null;
+
+            int attempt = 0;
+            while (attempt < maxAttempts)
+            {
+                attempt++;
+
+                outputHelper.WriteLine($"Attempt #{attempt} of {maxAttempts}: GET /processes");
+
+                identifiers = (await client.GetProcessesAsync()).ToList();
+
+                if (null != filter)
```

Comment:
Don't you want to put the filtering after the retries, since filtering by process name will be broken? (Although I didn't see any clients that do this)

Reply:
I wanted to filter to processes that we cared about rather than verifying that all of the processes we found have valid names. I can change the filter to just be on pid (or provide a list of pods instead of a filter) to force the intention rather that leaving it up to the tests.